src/lib.rs

```diff
@@ -146,14 +139,17 @@ where
     /// An execution error.
     type Error;
 
+    /// A `Seed`, suitable to generate a `Rng`.
+    type RngSeed;
+
     /// Execute an algorithm over a context and graph.
     /// Changes to the context will be persisted across
     /// executions of the algorithm.
-    fn execute<R: SeedableRng>(
+    fn execute(
         &self,
         context: &mut Self::Context,
         graph: &mut G,
-        rng: R,
+        initial_seed: Self::RngSeed,
```

Comment:
```suggestion
        seed: Self::RngSeed,
```
There's only one seed :seedling:  ;)